pandas/tests/indexing/test_loc.py

```diff
@@ -1106,3 +1107,137 @@ def test_loc_with_period_index_indexer():
     tm.assert_frame_equal(df, df.loc[list(idx)])
     tm.assert_frame_equal(df.iloc[0:5], df.loc[idx[0:5]])
     tm.assert_frame_equal(df, df.loc[list(idx)])
+
+
+def test_loc_setitem_df_datetime64tz_full_column_with_index():
+    df = pd.DataFrame(
+        pd.date_range("2020-01-01", "2020-01-06", 6, tz=timezone.utc), columns=["data"]
+    )
+    df2 = pd.DataFrame(index=df.index)
```

Comment:
use 

result = 
expected = 

as its not really clear what you are comparing here